src/nomad_simulations/variables.py

```diff
@@ -150,3 +155,74 @@ def __init__(
 
     def normalize(self, archive, logger) -> None:
         super().normalize(archive, logger)
+
+
+class KMesh(Variables):
+    """
+    K-point mesh over which the physical property is calculated. This is used to define `ElectronicEigenvalues(PhysicalProperty)` and
+    other k-space properties. The `points` are obtained from a refernece to the `NumericalSettings` section, `KMesh(NumericalSettings)`.
+    """
+
+    k_mesh_settings_ref = Quantity(
+        type=KMeshSettings,
+        description="""
+        Reference to the `KMesh(NumericalSettings)` section in the `ModelMethod` section. This reference is useful
+        to extract `points` and, then, obtain the shape of `value` of the `PhysicalProperty`.
+        """,
+    )
+
+    points = Quantity(
+        type=KMeshSettings.points,
+        description="""
+        Reference to the `KMesh.points` over which the physical property is calculated. These are 3D arrays stored in fractional coordinates.
+        """,
+    )
```

Comment:
Given the description in `k_mesh_settings_ref`, is this quantity really helpful?
I mean, now there are 2 ways of retrieving exactly the same property.